src/typescript/sdk/src/client/emojicoin-client.ts

```diff
@@ -0,0 +1,332 @@
+ *
+ * The `utils` functions provides several commonly used utility functions.
+ *
+ * @example
+ * ```typescript
+ * const emojis: MarketSymbolEmojis = ["🌊"];
+ * const emojicoin = new EmojicoinClient();
+ * const account = Account.generate();
+ * const integrator = account.accountAddress;
+ * await emojicoin.register(account, emojis, { integrator });
+ * const buyArgs = {
+ *  inputAmount: 100n,
+ *  minOutputAmount: 1n,
+ *  integrator,
+ *  integratorFeeRateBPs: 0,
+ * };
+ * await emojicoin.buy(account, emojis, buyArgs);
+ * ```
+ */
+export class EmojicoinClient {
+  public aptos: Aptos;
+
+  public liquidity = {
+    provide: this.provideLiquidity.bind(this),
+    remove: this.removeLiquidity.bind(this),
+  };
+
+  public utils = {
+    emojisToHexStrings: this.emojisToHexStrings.bind(this),
+    emojisToHexSymbol: this.emojisToHexSymbol.bind(this),
+    getEmojicoinInfo: this.getEmojicoinInfo.bind(this),
+    getTransactionEventData: this.getTransactionEventData.bind(this),
+  };
+
+  public rewards = {
+    buy: this.buyWithRewards.bind(this),
+    sell: this.sellWithRewards.bind(this),
+  };
+
+  constructor(args?: { aptos?: Aptos }) {
+    this.aptos = args?.aptos ?? getAptosClient().aptos;
+  }
+
+  async register(
+    registrant: Account,
+    emojis: MarketSymbolEmojis,
+    args: {
+      integrator: AccountAddressInput;
+    },
+    options?: Options
+  ) {
+    const response = await RegisterMarket.submit({
+      aptosConfig: this.aptos.config,
+      registrant,
+      emojis: this.emojisToHexStrings(emojis),
+      integrator: args.integrator,
+      ...options,
+    });
+    const res = this.getTransactionEventData(response);
+    return {
+      ...res,
+      registration: {
+        event: expect(res.events.marketRegistrationEvents.at(0), Expect.Register.Event),
+        model: expect(res.models.marketRegistrationEvents.at(0), Expect.Register.Model),
+      },
+    };
+  }
+
+  async chat(
+    user: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      chatEmojis: AnyEmoji[];
+      emojiIndicesSequence: number[];
+    },
+    options?: Options
+  ) {
+    const chatEmojis = getEmojisInString(args.chatEmojis.join("")) as SymbolEmoji[];
+    const emojiBytes = this.emojisToHexStrings(chatEmojis);
+    const emojiIndicesSequence = new Uint8Array(args.emojiIndicesSequence);
+
+    const response = await Chat.submit({
+      aptosConfig: this.aptos.config,
+      user,
+      emojiBytes,
+      emojiIndicesSequence,
+      ...options,
+      ...this.getEmojicoinInfo(emojis),
+    });
+    const res = this.getTransactionEventData(response);
+    return {
+      ...res,
+      chat: {
+        event: expect(res.events.chatEvents.at(0), Expect.Chat.Event),
+        model: expect(res.models.chatEvents.at(0), Expect.Chat.Model),
+      },
+    };
+  }
+
+  async buy(
+    swapper: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      inputAmount: Uint64;
+      minOutputAmount: Uint64;
```

Comment:
How about making this optional with a default value of either 99% or just 1. Since we want to use it for easy testing, it would be easier to pass fewer args, and we will rarely need to pass a minimum output amount.

Reply:
So actually my first instinct was to do that- I initially made the class with a bunch of default args and assumptions because it makes it even easier to use (like you're saying, fewer args, easier to read, etc).

But I think it's better if we keep this specifically as a class to simplify transaction submission without making any assumptions and make another class extended from this one specifically as a test helper. Partly to separate concerns (general helper class vs test-only helper class) and partly to make it easier to test.

I can add the test-only class today and that should cover streamlining some of the unnecessary abstractions in our tests